Utilities/build-using-self

```diff
@@ -107,22 +107,22 @@ def set_environment(*, swiftpm_bin_dir: pathlib.Path,) -> None:
 
 
 def run_bootstrap(swiftpm_bin_dir: pathlib.Path) -> None:
-    if is_on_darwin():
-        logging.info("Current working directory is %s", pathlib.Path.cwd())
-        logging.info("Bootstrapping with the XCBuild codepath...")
-        call(
-            [
-                REPO_ROOT_PATH / "Utilities" / "bootstrap",
-                "build",
-                "--release",
-                "--verbose",
-                "--cross-compile-hosts",
-                "macosx-arm64",
-                "--skip-cmake-bootstrap",
-                "--swift-build-path",
-                (swiftpm_bin_dir / "swift-build").resolve(),
-            ],
-        )
+    logging.info("Current working directory is %s", pathlib.Path.cwd())
+    logging.info("Bootstrapping with the XCBuild codepath...")
+    cross_compile_arch = "x86_64" if platform.machine() == "arm64" else "arm64"
```

Comment:
nit: would be nice to add an error case for a third architecture for future-proofing instead of assuming the else case wants arm64

Reply:
Good call. updated